### PR TITLE
add support for drag and drop of files into the file menu

### PR DESCRIPTION
**Pull request by seliayeu**

### CODE REVIEW

File: src/components/FileViewer/FileViewer.tsx [hunk 1]
```diff
@@ -218,33 +221,69 @@ export default function FileViewer() {
...
+  //   console.log(event);
+  //   for (const i in event.payload) {
+  //     const fileBlob = new Blob([await readBinaryFile(event.payload[i])]);
+  //     const file = new File([fileBlob], event.payload[i].split(directorySeparator).pop() ?? 'unknown');
+  //     fileArray.push(file);
+  //   }
+  //   console.log(fileArray);
+    
+  //   uploadInstanceFiles(instance.uuid, path, fileArray, queryClient);
+  // };
```
Comment (CheatCod) [inline src/components/FileViewer/FileViewer.tsx:259]: commented code

File: src/components/FileViewer/FileViewer.tsx [hunk 2]
```diff
@@ -320,7 +359,69 @@ export default function FileViewer() {
...
+          >
+            <div className="fixed inset-0 bg-[#000]/80" />
+            <div className="fixed inset-0 overflow-y-auto">
+              <div className="pointer-events-none flex min-h-full items-center justify-center p-4 text-center text-white/50">
+                <Dialog.Panel className="pointer-events-none  flex w-[200px] flex-col items-center justify-center gap-4 rounded-3xl border-2 border-dashed border-gray-faded/30 bg-gray-800 bg-opacity-50 pb-8 pt-12">
+                  <FontAwesomeIcon
+                    className="pointer-events-none m-0 p-0 text-h1 text-gray-faded/80"
+                    icon={faDownload}
+                  />
+                  <p className="pointer-events-none">drop file</p>
```
Comment (CheatCod) [inline src/components/FileViewer/FileViewer.tsx:419]: Change to more descriptive text like "Drop files to upload"

**Review by CheatCod — CHANGES_REQUESTED** [review]
Currently if someone tries to upload a folder it just toasts "Network Error", maybe display the overlay to "Cannot upload folders" when someone drags a folder over?

File: src/components/FileViewer/FileViewer.tsx [hunk 3]
```diff
@@ -59,6 +56,9 @@ import { useFileContent, useFileList } from 'data/FileSystem';
 import { FileEditor } from './FileEditor';
 import { UnzipOption } from 'bindings/UnzipOptions';
 
+import { listen } from '@tauri-apps/api/event';
```
Comment (Ynng) [inline src/components/FileViewer/FileViewer.tsx:59]: remove unused import?
I can't really find where this is being used

File: src/components/FileViewer/FileViewer.tsx [hunk 4]
```diff
@@ -218,33 +221,54 @@ export default function FileViewer() {
...
+
+    for (const i in event.dataTransfer.items) {
+        const item = event.dataTransfer.items[i];
+        if (item.kind !== "file") continue;
+        const entry = "getAsEntry" in DataTransferItem.prototype ? (item as any).getAsEntry() : item.webkitGetAsEntry();
+        if (entry.isDirectory) {
+          toast.error("Only files can be uploaded");
+          setDropping(false);
+          setDroppingDialog(false);
+          return;
```
Comment (Ynng) [inline src/components/FileViewer/FileViewer.tsx:259]: might be better to `continue` than return and make this a warning?

**Review by Ynng — CHANGES_REQUESTED** [review]

File: src/components/FileViewer/FileViewer.tsx (hunk 4 above)
Comment (seliayeu) [reply src/components/FileViewer/FileViewer.tsx:259]: not entirely sure what causes the behavior but this `continue` is pretty much only reached when uploading a normal file. at the start of the for loop, it sometimes yields an item with an undefined `kind` which causes the `webkitGetAsEntry()` to not work. not sure how else to do this part

**Review by CheatCod — APPROVED** [review]

**Comment by CheatCod** [comment]
imma merge if kevin hasa problem he can fight me

**Comment by Ynng** [comment]
wtf man not cool

**Pull request merged**